File: Downloads/TechTestJava-master/TechTestJava-master/anycompany/src/main/java/za.co.anycompany/service/OrderService.java
package za.co.anycompany.service;

import za.co.anycompany.datalayer.CustomerRepository;
import za.co.anycompany.datalayer.OrderRepository;
import za.co.anycompany.model.Customer;
import za.co.anycompany.model.Order;

public class OrderService {

    private OrderRepository orderRepository = new OrderRepository();

    public boolean placeOrder(Order order, int customerId)
    {
        Customer customer = CustomerRepository.load(customerId);
        if(customer != null) {
            if (order.getAmount() == 0)
                return false;

            if (customer.getCountry() == "UK")
                order.setVAT(0.2d);
            else
                order.setVAT(0);

            orderRepository.save(order);

            return true;
        }
        else{
            return false;
        }
    }
}

File: AnyCompanyEntertainment/AnyCompanyEntertainmentDotNet/Services/OrderService.cs
using AnyCompanyEntertainmentDotNet.Models;
using AnyCompanyEntertainmentDotNet.Services.Interfaces;
using AnyCompanyEntertainmentDotNet.Services.Repositories.Interfaces;

namespace AnyCompanyEntertainmentDotNet.Services
{
    public class OrderService : IOrderService
    {
        private readonly ICustomerRepository customerRepository;
        private readonly IOrderRepository orderRepository;
        private readonly AppDbContext dbContext;

        public OrderService(ICustomerRepository customerRepository, IOrderRepository orderRepository, AppDbContext dbContext)
        {
            this.customerRepository = customerRepository;
            this.orderRepository = orderRepository;
            this.dbContext = dbContext;
        }

        public void PlaceOrder(int customerId, Order order)
        {
            Customer? customer = customerRepository.GetById(customerId);
            if (customer != null)
            {
                order.CustomerId = customerId;
                orderRepository.Add(order);
                Console.WriteLine("Order placed successfully.");
            }
            else
            {
                Console.WriteLine("Customer not found.");
            }
        }

        public Customer? GetCustomerWithOrders(int customerId)
        {
            return customerRepository.GetById(customerId);
        }

        public List<Customer> GetAllCustomersWithOrders()
        {
            return customerRepository.GetAll();
        }
    }
}
